expression/builtin_miscellaneous.go

```diff
@@ -326,7 +327,54 @@ type builtinInet6AtonSig struct {
 
 // See https://dev.mysql.com/doc/refman/5.7/en/miscellaneous-functions.html#function_inet6-aton
 func (b *builtinInet6AtonSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("INET6_ATON")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	if args[0].IsNull() {
+		return d, nil
+	}
+
+	ipAddress, err := args[0].ToString()
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	if len(ipAddress) == 0 {
+		return d, nil
+	}
+
+	ip := net.ParseIP(ipAddress)
+	if ip == nil {
+		return d, nil
+	}
+
```

Comment:
we can not directly return cus golang treat mapped IPv6 address as an IPv4 address, see here:
https://github.com/golang/go/blob/master/src/net/ip_test.go#L22

But mysql treat it as IPv6 address:
<img width="407" alt="2017-04-19 9 46 31" src="https://cloud.githubusercontent.com/assets/6668939/25183329/d67fde82-2549-11e7-8211-8326a723e9d0.png">

So, we need some code to deal this situation.So mysql need `is_ipv4_mapped` to check this ip address.